server/rest/src/main/java/org/infinispan/rest/resources/BackupManagerResource.java

```diff
@@ -198,6 +198,8 @@ static CompletionStage<RestResponse> handleRestore(InvocationHelper invocationHe
                         Files.delete(path);
                      } catch (IOException e) {
                         LOG.warnf(e, "Unable to delete uploaded backup file '%s'", path);
+                     } finally {
+                        decoder.destroy();
```

Comment:
can `decoder` be `null`?

Reply:
Indeed, it can. We need to protect from it